obc/app/modules/command_mgr/command_callbacks.c

```
@@ -84,19 +84,23 @@ obc_error_code_t pingCmdCallback(cmd_msg_t *cmd) {
   return OBC_ERR_CODE_SUCCESS;
 }
 
-obc_error_code_t downlinkTelemCmdCallback(cmd_msg_t *cmd) {
+static obc_error_code_t downlinkTelemCmdCallback(cmd_msg_t *cmd) {
   obc_error_code_t errCode;
 
   RETURN_IF_ERROR_CODE(setTelemetryManagerDownlinkReady());
 
   return OBC_ERR_CODE_SUCCESS;
 }
 
-obc_error_code_t uplinkDisconnectCmdCallback(cmd_msg_t *cmd) {
-  obc_error_code_t errCode;
-
-  comms_event_t disconnectSentEvent = {.eventID = COMMS_EVENT_START_DISC};
-  RETURN_IF_ERROR_CODE(sendToCommsManagerQueue(&disconnectSentEvent));
-
-  return OBC_ERR_CODE_SUCCESS;
-}
+const cmd_info_t cmdsConfig[] = {
+    [CMD_END_OF_FRAME] = {NULL, CMD_POLICY_PROD, CMD_TYPE_NORMAL},
+    [CMD_EXEC_OBC_RESET] = {execObcResetCmdCallback, CMD_POLICY_PROD, CMD_TYPE_CRITICAL},
+    [CMD_RTC_SYNC] = {rtcSyncCmdCallback, CMD_POLICY_PROD, CMD_TYPE_NORMAL},
+    [CMD_DOWNLINK_LOGS_NEXT_PASS] = {downlinkLogsNextPassCmdCallback, CMD_POLICY_PROD, CMD_TYPE_CRITICAL},
+    [CMD_MICRO_SD_FORMAT] = {microSDFormatCmdCallback, CMD_POLICY_PROD, CMD_TYPE_CRITICAL},
+    [CMD_PING] = {pingCmdCallback, CMD_POLICY_PROD, CMD_TYPE_NORMAL},
+    [CMD_DOWNLINK_TELEM] = {downlinkTelemCmdCallback, CMD_POLICY_PROD, CMD_TYPE_NORMAL}};
+
+// This function is purely to trick the compiler into thinking we are using the cmdsConfig variable so we avoid the
+// unused variable error
+void unusedFunc() { UNUSED(cmdsConfig); }
```

`static`

---

Adding static gives an unused function error, is there a way to get around this?

---

From the meeting we talked about adding this to the verify command function but that does not take away the error from this file. I can't add it to any of the call backs since the variable needs to be defined after the callbacks. Is there any way to make the compiler ignore the error?

---

okay its fine to leave this here then, there's another way to fix it but its more work and not worth it